Source/SwiftLintFramework/Rules/Idiomatic/VoidFunctionInTernaryConditionRule.swift

```diff
@@ -0,0 +1,235 @@
+                index == 0 ↓? something() : somethingElse(index)
+                return index
+            """),
+            Example("""
+            var value: Int {
+                mode == .fast ↓? something() : somethingElse()
+                return 0
+            }
+            """),
+            Example("""
+            var value: Int {
+                get {
+                    mode == .fast ↓? something() : somethingElse()
+                    return 0
+                }
+            }
+            """),
+            Example("""
+            subscript(index: Int) -> Int {
+                get {
+                    index == 0 ↓? something() : somethingElse(index)
+                    return index
+                }
+            """)
+        ]
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        let visitor = VoidFunctionInTernaryConditionVisitor()
+        return visitor.walk(file: file) { visitor in
+            visitor.violations(for: self, in: file)
+        }
+    }
+}
+
+private class VoidFunctionInTernaryConditionVisitor: SyntaxVisitor {
+    private var positions = [AbsolutePosition]()
+
+    override func visitPost(_ node: TernaryExprSyntax) {
+        guard node.firstChoice.is(FunctionCallExprSyntax.self),
+              node.secondChoice.is(FunctionCallExprSyntax.self),
+              let parent = node.parent?.as(ExprListSyntax.self),
+              !parent.containsAssignment,
+              let grandparent = parent.parent,
+              grandparent.is(SequenceExprSyntax.self),
+              let blockItem = grandparent.parent?.as(CodeBlockItemSyntax.self),
+              !blockItem.isImplicitReturn else {
+            return
+        }
+
+        positions.append(node.questionMark.positionAfterSkippingLeadingTrivia)
+    }
+
+    func violations(for rule: VoidFunctionInTernaryConditionRule, in file: SwiftLintFile) -> [StyleViolation] {
+        return positions.map { position in
+            StyleViolation(ruleDescription: type(of: rule).description,
+                           severity: rule.configuration.severity,
+                           location: Location(file: file, byteOffset: ByteCount(position)))
+        }
+    }
+}
+
+private extension ExprListSyntax {
+    var containsAssignment: Bool {
+        return children.contains(where: { $0.is(AssignmentExprSyntax.self) })
+    }
+}
+
+private extension CodeBlockItemSyntax {
+    var isImplicitReturn: Bool {
+        isClosureImplictReturn || isFunctionImplicitReturn ||
+        isVariableImplicitReturn || isSubscriptImplicitReturn ||
+        isAcessorImplicitReturn
+    }
+
+    var isClosureImplictReturn: Bool {
+        guard let parent = parent?.as(CodeBlockItemListSyntax.self),
+              let grandparent = parent.parent else {
+            return false
+        }
+
+        return parent.children.count == 1 && grandparent.is(ClosureExprSyntax.self)
+    }
+
+    var isFunctionImplicitReturn: Bool {
```

Comment:
The blocks of the computed variables `isFunctionImplicitReturn`, `isVariableImplicitReturn`, `isSubscriptImplicitReturn` and `isAcessorImplicitReturn` are quite similar and could be replace by calls to the method
```swift
private func check<T: SyntaxProtocol>(for type: T.Type, condition: (T) -> Bool = { _ in true }) -> Bool {
    guard let parent = parent?.as(CodeBlockItemListSyntax.self),
          let grandparent = parent.parent?.parent?.as(type) else {
        return false
    }
    return parent.children.count == 1 && condition(grandparent)
}
```
to avoid some code duplication.

Reply:
That's clever! However I think that the code duplication here is small enough that using a generic function with a closure as condition would make it harder to follow. We can always refactor this in the future if we notice we're making changes to all these variables